rflx/expression.py

```diff
@@ -40,26 +40,33 @@ class ProofResult(Enum):
 
 
 class Proof:
+    LOGIC = "QF_NIA"
+
     def __init__(self, expr: "Expr", facts: Optional[Sequence["Expr"]] = None):
         self.__expr = expr
         self.__facts = facts or []
         self.__result = ProofResult.UNSAT
+        self.__unknown_reason = "unknown"
 
-        solver = z3.Solver()
+        solver = z3.SolverFor(self.LOGIC)
         solver.add(self.__expr.z3expr())
         for f in self.__facts:
             solver.add(f.z3expr())
 
         self.__result = ProofResult(solver.check())
+        if self.__result == ProofResult.UNKNOWN:
+            self.__unknown_reason = solver.reason_unknown()
```

Comment:
Couldn't this be done in the `error` property? The unsat core is also analyzed there. Why should `UNKOWN` handled differently than `UNSAT`?

Reply:
For `UNSAT` we rerun the proof completely with unsat core generation enabled (as the original solver object just doesn't have this information). Do we really want to rerun the proof for `UNKNOWN` just to get the same information we already had in the previous proof attempt?

Reply:
No, this doesn't make sense. I didn't realize that rerunning the proof would be necessary.